webapp/channels/src/components/announcement_bar/users_limits_announcement_bar/index.tsx

```diff
@@ -0,0 +1,103 @@
+import AnnouncementBar from 'components/announcement_bar/default_announcement_bar';
+import useCWSAvailabilityCheck, {CSWAvailabilityCheckTypes} from 'components/common/hooks/useCWSAvailabilityCheck';
+
+import {AnnouncementBarTypes, LicenseLinks, ModalIdentifiers} from 'utils/constants';
+
+type Props = {
+    license?: ClientLicense;
+};
+
+function UsersLimitsAnnouncementBar(props: Props) {
+    const dispatch = useDispatch();
+
+    const {formatMessage} = useIntl();
+
+    const cwsAvailability = useCWSAvailabilityCheck();
+    const usersLimits = useSelector(getUsersLimits);
+
+    const handleCTAClick = useCallback(() => {
+        if (cwsAvailability === CSWAvailabilityCheckTypes.Available || cwsAvailability === CSWAvailabilityCheckTypes.Unknown) {
+            window.open(LicenseLinks.CONTACT_SALES, '_blank');
+        } else if (cwsAvailability === CSWAvailabilityCheckTypes.Unavailable) {
+            // Its an airgapped instance
+            dispatch(openModal({
+                modalId: ModalIdentifiers.AIR_GAPPED_CONTACT_SALES,
+                dialogType: AirGappedContactSalesModal,
+            }));
+        }
+    }, [cwsAvailability]);
+
+    const maxUsersLimit = usersLimits?.maxUsersLimit ?? 0;
+    const activeUserCount = usersLimits?.activeUserCount ?? 0;
+    const isLicensed = props?.license?.IsLicensed === 'true';
+    const isCloud = props?.license?.Cloud === 'true';
+    const isTrial = props?.license?.IsTrial === 'true';
+
+    if (!shouldShowUserLimitsAnnouncementBar({isLicensed, isCloud, isTrial, maxUsersLimit, activeUserCount})) {
+        return null;
+    }
+
+    const copyText = formatMessage({
+        id: 'users_limits_announcement_bar.copyText',
+        defaultMessage: 'Your user count exceeds the maximum users allowed. Upgrade to Mattermost Professional or Mattermost Enterprise to continue using Mattermost.',
+    });
+    const ctaText = formatMessage({
+        id: 'users_limits_announcement_bar.ctaText',
+        defaultMessage: 'Contact sales',
+    });
+
+    return (
+        <AnnouncementBar
+            id='users_limits_announcement_bar'
+            showCloseButton={false}
+            message={copyText}
+            type={AnnouncementBarTypes.CRITICAL}
+            icon={<ExclamationThickIcon size={16}/>} // Icon to be fa-exclamation-triangle
+            showCTA={true}
+            ctaDisabled={cwsAvailability === CSWAvailabilityCheckTypes.Loading}
+            showLinkAsButton={true}
+            ctaText={ctaText}
+            onButtonClick={handleCTAClick}
+        />
+    );
+}
+
+export type ShouldShowingUserLimitsAnnouncementBarProps = {
+    isLicensed: boolean;
+    isCloud: boolean;
+    isTrial: boolean;
+    maxUsersLimit: number;
+    activeUserCount: number;
+};
+
+export function shouldShowUserLimitsAnnouncementBar({isLicensed, isCloud, isTrial, maxUsersLimit, activeUserCount}: ShouldShowingUserLimitsAnnouncementBarProps) {
+    if (maxUsersLimit === 0 || activeUserCount === 0) {
```

Comment:
Why do we want to check here for `activeUserCount === 0`? The condition doesn't hurt here and probably a case that wont ever happen but not sure how it is relevant here

Reply:
API return both of them 0 for licensed, so i thought its good to have that here as well

Reply:
I see, makes sense